#FT-1 ft(design): home page design

chrisostome01 opened the pull request:
# PR Description

This Pull Request aims to add design for the landing page of the Medical core

# Description of what this task covered

- [x]  CSS and HTML only
- [x]  BEM methodology

# How this may be tested?

1. Clone the repo
4. Navigate into crispy-t 
3. Check out to the appropriate branch(ft/design-home-page)
4. Open index.html in your browser or you can use your live server extension


# Screenshots (If appropriate/applicable)
<img width="950" alt="2022-09-15" src="https://user-images.githubusercontent.com/56281095/190918177-52cba9fe-9eb2-4d8c-bfcb-48b4cce870c0.png">

<img width="951" alt="2022-09-15 (1)" src="https://user-images.githubusercontent.com/56281095/190918184-fccb1816-7f11-4ede-ab1b-47398aaf7f70.png">

<img width="951" alt="2022-09-15 (2)" src="https://user-images.githubusercontent.com/56281095/190918187-9ab21656-d197-4774-a1a4-a8d0d2e9e9ea.png">

<img width="951" alt="2022-09-15 (3)" src="https://user-images.githubusercontent.com/56281095/190918195-e5abf4d8-5f47-4477-bb47-404bcf97b06f.png">

chrisostome01 committed:
ft(design): home page design

Create header section
Creates main content section
Adds team cards section
Adds footer design
Adds subscription input

#done

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                    Medical Core
+                </h1>
+            </div>
+            <ul class="nav-link" >
```
great job making the list a block!

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+        <img class="content__image" src="./assets/img/n.png" alt="" srcset="">
+        <div class="content__description">
+            <div class="content__line content__line--red "></div>
+            <div class="site-heading site-heading--color-main site-heading--spaced">Meet our Experts</div>
```
i don't think the site heading should be  a block

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+        </div>
+        <div class="team-cards">
+            <!-- user 1  -->
+            <div class="card card__size-small card--shadow card--rounded">
```
not a good naming : .card__size-small     size-small is not a good name to give an element.

or did you accidentally make it an element  instead of a modifier

chrisostome01 replied:
Oooh let me change.

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                </div>
+            </div>
+            <!-- user 2 -->
+            <div class="card card__size-small card--shadow card--rounded">
```
same here

chrisostome01 replied:
Because they are the same card I think both have the same issue. I will work on that too

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                    Proffessional
+                </div>
+                <div class="card__icons card__icons--center">
+                    <svg width="23" height="23" viewBox="0 0 23 23" fill="none" xmlns="http://www.w3.org/2000/svg">
```
why a lot of lines for just 3 icons??, why not import the pictures from figma or just use Fa-fa 🤷‍♀️🤷‍♀️

-------P.S Not a review -----

chrisostome01 replied:
I think we can use img tag instead of font awesome. because we are  restricted to use any CSS library

alain-kubwayo replied:
You're right. Introducing Font Awesome would go against sticking to only HTML and CSS for the project. SVGs should be fine, I think.

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+    <!-- ============= End: Team card ========= -->
+    
+    <!-- ========== Start: footer ========= -->
+    <div class="footer footer--full footer--secondary footer--alignment-center">
```
too much modifiers when a footer is just going to be used once!

alain-kubwayo replied:
Way too many modifiers for a one-time use block.

serapieTuyishime commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                the two major realms of Classical physics: Newtonian mechanics 
+            </p>
+        </div>
+        <form class="newsletter newsletter--size-full newsletter--position-astray newsletter--align-center">
```
modifiers

sezeranoJchrisostome replied:
I did not get what you meant about modifiers please @serapieTuyishime

serapieTuyishime commented on styles/main.css:
```diff
@@ -0,0 +1,376 @@
+
+
+/* start:navigation__button */
+.navigation__button {
```
you can just delete this since it is not styling anything anyway

alain-kubwayo replied:
Always a good idea to delete things like classes/styles that are not doing anything in our code.

serapieTuyishime commented on styles/main.css:
```diff
@@ -0,0 +1,376 @@
+    border: 1px solid #E6E6E6;
+}
+
+/* end newsletter */
```
Is it responsive without using the media queries?

serapieTuyishime reviewed (changes requested):
Not much to change!

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                    Receive care
+                </p>
+            </div>
+            <div class="header__text">
+                <p class="header__description">
+                    Get your best looking smile now!
+                </p>
+                <button class="button button--success button--rounded button--medium button--bold">
+                    <span class="button__text ">
+                        Join Us
+                    </span>
+                    <span class="button__icon" >
+                        <img src="./assets/svg/left-icon.svg" alt="">
+                    </span>
+                </button>
+                <button class="button button--success-outline button--rounded button--medium button--bold">
+                    <span class="button__text ">
+                        Learn More
+                    </span>
+                </button>
+            </div>
```
Excellent job following BEM naming convention!

chrisostome01 replied:
Thank you! @alain-kubwayo

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+   
+    <!-- ========= Start: Content ========= -->
+    <div class="content">
+        <img class="content__image" src="./assets/img/n.png" alt="" srcset="">
```
It's always a good practice to give images alt value in case the server can't display the image correctly.

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                    Julian Jameson
+                </div>
+                <div class="card__text card__text--center card__text--color-main  card__text--size-small">
+                    Proffessional
```
You made a minor typo here and on the other cards as well. It should be `profession` instead of `proffessional`

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+                    Julian Jameson
+                </div>
+                <div class="card__text card__text--center card__text--color-main  card__text--size-small">
+                    Proffessional
```
Typo

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+            Newsletter
+        </div>
+        <div class="site-heading site-heading--color-main site-heading--spaced-small">JOIN US</div>
+        <div class="footer__text footer__text--gray">
+            <p>
+                Problems trying to resolve the conflict between <br/>
+                the two major realms of Classical physics: Newtonian mechanics 
+            </p>
+        </div>
```
This looks okay but I would personally add `class="footer__text footer__text--gray"` to `<p>` tag to avoid introducing extra `<div>` unnecessarily

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+            </p>
+        </div>
+        <form class="newsletter newsletter--size-full newsletter--position-astray newsletter--align-center">
+            <input placeholder="Your Email" type="text" class="newsletter__field newsletter__field--text-medium newsletter__field--text-gray newsletter__field--size-full newsletter__field--rounded-left">
```
Since you only have one input throughout your application, I think there are way too many modifiers for that. A lot of the styles could be added to the element `newsletter__field` for simplicity.

alain-kubwayo commented on index.html:
```diff
@@ -0,0 +1,231 @@
+        </div>
+        <form class="newsletter newsletter--size-full newsletter--position-astray newsletter--align-center">
+            <input placeholder="Your Email" type="text" class="newsletter__field newsletter__field--text-medium newsletter__field--text-gray newsletter__field--size-full newsletter__field--rounded-left">
+            <button type="submit" class="newsletter__button newsletter__button--color-success newsletter__button--rounded-right">Subscribe</button>
```
I believe you can reuse the `login` or `join us` button block for this instead of creating a brand new button from scratch.

chrisostome01 replied:
Thank you! @alain-kubwayo, I will try to reuse it and I will let you know if it is going to work for me

alain-kubwayo reviewed (commented):
Excellent work! You adhered to BEM naming convention throughout the project. I have added some comments on the areas that need fine-tuning for your reference. Cheers!

chrisostome01 committed:
perf: apply requested changes

Removes unnecessaily modifiers
Adds responsiveness
Refactor svgs to imgs
Correcting element name

#done

serapieTuyishime reviewed (approved):
Well done mate!

alain-kubwayo reviewed (approved):
LGTM

karemarsy commented on styles/main.css:
```diff
@@ -0,0 +1,364 @@
+/* end: Navigation */
+
+/* start:nav-link */
+.nav-link{
```
why did you create another class called `nav-link` while you already have `navigation`? you should use `nav-link` as an element of the `navigation` that would make sense

karemarsy commented on styles/main.css:
```diff
@@ -0,0 +1,364 @@
+    justify-content: space-between;
+    gap: 5px;
+}
+.card__image{
```
where does this class call `card` comes from?

chrisostome01 replied:
I defined it in `index.html` but I can try to add it in my CSS too but they will be not much style to add there.

serapieTuyishime replied:
Nice ✔

karemarsy commented on styles/main.css:
```diff
@@ -0,0 +1,364 @@
+    text-align: center;
+    gap: 9px;
+}
+.footer.footer--secondary {
```
That's not BEM

chrisostome01 replied:
I had confusion about that but now I will stick to the BEM tree and use `footer--secondary`

karemarsy commented on styles/responsiveness.css:
```diff
@@ -0,0 +1,77 @@
+
+
+/* start: content responsiveness */
+@media only screen and (max-width: 600px) {
```
why did you duplicate this media query you already have `@media only screen and (max-width: 600px)` above

chrisostome01 replied:
I was trying to put things together in my CSS, like content responsiveness alone  in it's media query, but now I can change to same media query since there the same

chrisostome01 committed:
perf: apply @karemarsy review

Removes unnecessary media queries
Improves BEM tree in in css

#done

serapieTuyishime commented on styles/main.css:
```diff
@@ -0,0 +1,366 @@
+    text-align: center;
+    gap: 9px;
+}
+.footer--secondary {
```
great change!

serapieTuyishime replied:
Great change!

serapieTuyishime reviewed (approved):
Well resolved ✨

alain-kubwayo commented on styles/main.css:
```diff
@@ -298,7 +300,7 @@ body {
     gap: 9px;
 }
-.footer.footer--secondary {
+.footer--secondary {
```
Well-done on applying requested changes.

alain-kubwayo commented on index.html:
```diff
@@ -21,17 +21,17 @@ <h1 class="logo__text logo__text--color-main">
                     Pricing
                 </li>
-                <li class="nav-link__item">
+                <li class="navigation__list__item">
```
I didn't know this kind of naming convention existed in BEM but well-done applying the changes per @karemarsy's feedback.

chrisostome01 replied:
Thank you @alain-kubwayo,
yesterday we had a session and we discussed about nesting elements and I saw the possibility of how this could be possible and I guess that here we have to be careful just in case we have many elements that are nested. 

@karemarsy

chrisostome01 replied:
This type of nesting was allowed in the other version of BEM  but now is not supported and it is not good practice too, you can refer to this blog [Here](https://scalablecss.com/bem-nesting-grandchild-elements/), I also have updated myself to it

alain-kubwayo reviewed (approved):
Good job applying @karemarsy feedback.

karemarsy commented on styles/main.css:
```diff
@@ -0,0 +1,366 @@
+    width: 100px;
+}
+
+
```
remove the space here

sezeranoJchrisostome commented:
Merging crisp landing page
